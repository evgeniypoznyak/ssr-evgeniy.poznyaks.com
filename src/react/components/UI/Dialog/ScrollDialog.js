import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import {makeStyles} from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
    title: {
        flexGrow: 1,
    },
    content: {
        padding: 10,
    },
}));

export default function ScrollDialog(props) {
    const classes = useStyles();
    const [open, setOpen] = React.useState(false);
    const [scroll, setScroll] = React.useState('paper');

    const handleClickOpen = scrollType => () => {
        setOpen(true);
        setScroll(scrollType);
    };

    function handleClose() {
        setOpen(false);
    }

    return (
        <div>
            <Button onClick={handleClickOpen('paper')}>Read more</Button>
            <Dialog
                open={open}
                onClose={handleClose}
                scroll={scroll}
                aria-labelledby="scroll-dialog-title"
                fullWidth={true}
                maxWidth={'lg'}
            >
                <DialogTitle id="scroll-dialog-title">
                    {props.title}
                </DialogTitle>
                <DialogContent dividers={true} className={classes.content}>
                    <DialogContent className={classes.content}>
                        {props.children}
                    </DialogContent>
                </DialogContent>
                <DialogActions>
                    <Button onClick={handleClose} color="primary">
                        Close
                    </Button>
                </DialogActions>
            </Dialog>
        </div>
    );
}
