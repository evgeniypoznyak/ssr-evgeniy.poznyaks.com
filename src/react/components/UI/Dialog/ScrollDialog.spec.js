import React from 'react';
import {mount, shallow} from 'enzyme';
import ScrollDialog from './ScrollDialog';
import {Button, Dialog} from '@material-ui/core';

let wrapper;
let shallowWrapper;

beforeAll(() => {
    wrapper = mount(<ScrollDialog title={'testTitle123'}>testTitle456</ScrollDialog>);
    shallowWrapper = shallow(<ScrollDialog title={'test title'}>Some Text</ScrollDialog>);
});

describe('ScrollDialog', () => {
    it('should render correctly with enzyme', () => {
        expect(shallowWrapper.children.length).toBe(1);
        shallowWrapper.find(Button).at(0).simulate('click');
        expect(shallowWrapper.getElement()).toMatchSnapshot();
    });

    it('should render modal window with expected text and title if read more is clicked', () => {
        const button = wrapper.find(Button);
        button.simulate('click');
        const dialog = wrapper.find(Dialog);
        expect(wrapper.text()).toMatch('Read more');
        expect(dialog.text()).toMatch('testTitle123');
        expect(dialog.text()).toMatch('testTitle456');
    });

    it('should Dialog props["open"] toBeTruthy', () => {
        const button = wrapper.find(Button).at(0);
        button.simulate('click');
        const dialog = wrapper.find(Dialog);
        expect(dialog.props()['open']).toBeTruthy();
    });
});
