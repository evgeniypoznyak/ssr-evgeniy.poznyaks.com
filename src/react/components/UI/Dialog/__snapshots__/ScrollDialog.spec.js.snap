// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollDialog should render correctly with enzyme 1`] = `
<div>
  <ForwardRef(WithStyles)
    onClick={[Function]}
  >
    Read more
  </ForwardRef(WithStyles)>
  <ForwardRef(WithStyles)
    aria-labelledby="scroll-dialog-title"
    fullWidth={true}
    maxWidth="lg"
    onClose={[Function]}
    open={true}
    scroll="paper"
  >
    <ForwardRef(WithStyles)
      id="scroll-dialog-title"
    >
      test title
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)
      className="makeStyles-content-2"
      dividers={true}
    >
      <ForwardRef(WithStyles)
        className="makeStyles-content-2"
      >
        Some Text
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)
        color="primary"
        onClick={[Function]}
      >
        Close
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
  </ForwardRef(WithStyles)>
</div>
`;
